apps/arweave/src/ar_node_worker.erl

```diff
@@ -321,19 +323,24 @@ move_tx_to_mining_pool(StateIn, TX, GS) ->
 
 drop_waiting_txs(#{ waiting_txs := WaitingTXs }, DropTXs, GS) ->
 	{NewGS, _} = ar_gossip:send(GS, {drop_waiting_txs, DropTXs}),
+	TXs = sets:subtract(WaitingTXs, sets:from_list(DropTXs)),
 	{ok, [
-		{waiting_txs, sets:subtract(WaitingTXs, sets:from_list(DropTXs))},
-		{gossip, NewGS}
+		{waiting_txs, TXs},
+		{gossip, NewGS},
+		{mempool_size, sets:fold(fun(TX, Acc) -> Acc + ?TX_SIZE_BASE + byte_size(TX#tx.data) end, 0, TXs)}
```

Comment:
`DropTXs` should be smaller than `TXs` most of the time so it's better to compute the total size of `DropTXs` and subtract it from the current size.

Reply:
Done